### PR TITLE
MAINT: Enable more ruff rules

### PR DESCRIPTION
**Pull request by larsoner**
### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

Get our formatting rules up to date with MNE-Python. We should merge then add to `.git-blame-revs-ignore` (can do in #837 since that will probably take longer to get merged).

### CODE REVIEW

File: mne_bids_pipeline/_config.py [hunk 1]
```diff
@@ -101,7 +100,7 @@
 BIDS dataset.
 """
 
-exclude_runs: Optional[Dict[str, List[str]]] = None
+exclude_runs: Optional[dict[str, list[str]]] = None
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:103]: I think this is only supported in Python 3.10+

Comment (larsoner) [reply mne_bids_pipeline/_config.py:103]: Looks like 3.9+ I think which we have in our reqs?

https://docs.python.org/3.9/library/typing.html#type-aliases

Comment (larsoner) [reply mne_bids_pipeline/_config.py:103]: I think I found the PEP that says 3.9 https://peps.python.org/pep-0585/

Comment (hoechenberger) [reply mne_bids_pipeline/_config.py:103]: Oh, great! 😊 all good then!

**Pull request merged**